account_asset_management/models/account_move.py

```diff
@@ -124,9 +119,11 @@ def _reverse_move_vals(self, default_values, cancel=True):
         if move_vals["type"] not in ("out_invoice", "out_refund"):
             for line_command in move_vals.get("line_ids", []):
                 line_vals = line_command[2]  # (0, 0, {...})
-                asset = self.env["account.asset"].browse(line_vals["asset_id"])
-                asset.unlink()
-                line_vals.update(asset_profile_id=False, asset_id=False)
+                asset_id = line_vals.get("asset_id")
+                if asset_id:
+                    asset = self.env["account.asset"].browse(asset_id)
+                    asset.unlink()
```

Comment:
IIRC unlink is one of the things that flushes to DB, right?

If so, it might help to gather all IDs together and unlink them in batch.